10up/ElasticPress
# Add a class to the search form when autosuggestions are loading. Addresses #1610.

### Description of the Change

Adds an `.is-loading` class to the search form form of any autosuggestion-enabled search input when autosuggestions are loading.

### Alternate Designs

The class could be added to the input element only, or any of the ancestors. I chose the `<form>` element as it would allow the user to style the entire form during loading, including any custom elements they might add for a loading indicator.

### Verification Process

- Install and configure ElasticPress with autosuggest enabled.
- Add a search widget somewhere on a page.
- Inspect the HTML of the search form.
- Testing results  will work best if network speed is throttled.
- Begin typing in the search form, and observe that an `is-loading` class is added to the parent `<form>` element while suggestions are loading, and is removed once they have loaded.

### Checklist:






### Applicable Issues

 #1610

### Changelog Entry

Added an `is-loading` class to the search form while autosuggestions are loading.
